src/ape/api/providers.py

```diff
@@ -637,17 +637,17 @@ def client_version(self) -> str:
 
     @property
     def base_fee(self) -> int:
-        block = self.get_block("latest")
-        if not hasattr(block, "base_fee"):
-            raise APINotImplementedError("No base fee found in block.")
+        fee_history = self.web3.eth.fee_history(1, self.get_block("latest").number)
+        if len(fee_history["baseFeePerGas"]) < 2:
```

Comment:
Won't this RPC call fail on the previous line if it's unimplemented? Probably needs a test in both scenarios, try it on BSC just to see

Reply:
EthTester does not implement the RPC and ape-geth does.
I tried EthTester using this implementation. With my new changes (see my commits), it now resorts to the old way upon receiving the RPC not implemented error.

(However, since we know EthTester doesnt support it, I just made it short-circuit to the old way right away for perf).

I tested this out on ape-geth as well! It uses the new API which is pretty awesome.
So if you use BSC, it works like my first test where it attempts to use this but fails and falls back to the way that works. Ideally, we do the same for ape-bsc that we do for ape-test by making it short-circuit and not bother.

So we should be good now; bases covered and the thing will resort to the old way when it has to. When possible, plugins should provider their own way when they dont support it.